src/dodal/devices/thawer.py

```diff
@@ -1,15 +1,53 @@
+from asyncio import Task, create_task, sleep
 from enum import Enum
 
-from ophyd_async.core import StandardReadable
+from bluesky.protocols import Stoppable
+from ophyd_async.core import (
+    AsyncStatus,
+    Device,
+    SignalRW,
+    StandardReadable,
+)
 from ophyd_async.epics.signal import epics_signal_rw
 
 
+class ThawingException(Exception):
+    pass
+
+
 class ThawerStates(str, Enum):
     OFF = "Off"
     ON = "On"
 
 
-class Thawer(StandardReadable):
+class ThawingTimer(Device):
+    def __init__(self, control_signal: SignalRW[ThawerStates]) -> None:
+        super().__init__("thawing_timer")
+        self._control_signal = control_signal
+        self._thawing_task: Task | None = None
+
+    @AsyncStatus.wrap
+    async def set(self, time_to_thaw_for: float):
+        await self._control_signal.set(ThawerStates.ON)
+        if self._thawing_task and not self._thawing_task.done():
+            raise ThawingException("Thawing task already in progress")
```

Comment:
I don't think so. `catch_exception_and_warn` is for when we have an error that's sort of expected in UDC. Like no pin is not that unusual, it means we shouldn't try and carry on on the pin but we otherwise it's fine. If we hit `ThawingException` it will be because something unexpected is going on so stopping is probably good